File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include<unordered_map>
#include <vector>

using namespace std;

int solution(vector<vector<string>> clothes) {
	int answer = 1;
	unordered_map<string, int> cloth;

	for (int i = 0; i < clothes.size(); i++) {
		auto temp = cloth.find(clothes[i][1]);
		if (temp != cloth.end()) {
			temp->second = temp->second + 1;
		}
        else {
			cloth[clothes[i][1]] = 1;
		}
	}

	for (const auto& pair:cloth) {
		answer *= (pair.second + 1);
	}
	answer--;
	return answer;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer =1; 
        Map<String,Integer> map = new HashMap<>();
        for(String[] s: clothes){
            map.put(s[1], map.getOrDefault(s[1],0)+1);
        }
        for(int num:map.values()){
            answer *= (num+1);
        }

        return answer-1;
    }
}